threatresponse/api/entity.py

```
@@ -8,56 +10,46 @@ def __init__(self, request, url):
 
         self._url = url
 
-    def get(self, id_=None, response_type='json', **params):
+    def get(self, id_=None, **kwargs):
         if id_:
-            url = '%s/%s' % (self._url, id_)
+            url = '%s/%s' % (self._url, quote(id_))
         else:
             url = self._url
 
-        return self._get(
-            url,
-            params=params,
-            response_type=response_type
-        )
+        return self._get(url, **kwargs)
 
-    def post(self, payload, response_type='json',  **params):
-        return self._post(
-            self._url,
-            json=payload,
-            params=params,
-            response_type=response_type
-        )
+    def post(self, payload, **kwargs):
+        return self._post(self._url, json=payload, **kwargs)
 
-    def put(self, id_, payload, response_type='json'):
+    def put(self, id_, payload, **kwargs):
         return self._put(
-            '%s/%s' % (self._url, id_),
+            '%s/%s' % (self._url, quote(id_)),
             json=payload,
-            response_type=response_type
+            **kwargs
         )
 
-    def patch(self, id_, payload, response_type='json'):
+    def patch(self, id_, payload, **kwargs):
         return self._patch(
-            '%s/%s' % (self._url, id_),
+            '%s/%s' % (self._url, quote(id_)),
             json=payload,
-            response_type=response_type
+            **kwargs
         )
 
-    def delete(self, id_):
+    def delete(self, id_, **kwargs):
         self._delete(
-            '%s/%s' % (self._url, id_),
-            response_type='raw'
+            '%s/%s' % (self._url, quote(id_)),
+            response_type='raw',
```

This line can lead to errors in the future when a user explicitly calls `delete(id_, response_type='anything')`. In that case, there will be exceptions like `TypeError: got multiple values for keyword argument 'response_type'`. So first of all, I think it's safer to update `kwargs` overwriting `response_type` before passing to `_delete`, i.e.
```python
kwargs['response_type'] = 'raw'
```
On the other hand, I think it's worth adding `return`, because the actual result comes from the internal `_delete` method (it may be a plain response object or just `None`), so explicit `return` is better than just discarding any results.

---

I doubt that implicitly overwriting the value of a parameter is a good idea. I'm sure I'd be surprised if `delete(id_, response_type='json')` would somehow return a raw response object instead of parsed JSON. Thus, in my opinion, it's better to raise an error. But I agree, that it would probably be better to raise some meaningful error so that a user wouldn't be perplexed by seeing `TypeError: got multiple values for keyword argument 'response_type'`.